Receiver/BatteryStreamReceiver/BatteryStreamProcessing.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace BatteryStreamReceiver
+{
+    public class BatteryStreamProcessing : IProcessStreamingData
+    {
+        private BatteryStatastics batteryStatastics;
+        private IStreamInputParser _streamInputParser;
+        public BatteryStreamProcessing(IStreamInputParser streamInputParser)
+        {
+            _streamInputParser = streamInputParser;
+            batteryStatastics = new BatteryStatastics();
+
+        }
+        public BatteryStatastics CalculateAverageParametersReadings(List<string> batteryStreamingData)
+        {
+            List<BatteryParameters> batteryParameters = new List<BatteryParameters>();
+            batteryParameters = _streamInputParser.GetBatteryParametersFromStreamingData(batteryStreamingData);            
+            if (ListValidator.IsBatteryParameterListValid(batteryParameters))
+            {
+                double sumTemperatureValue = 0, sumStateOfChargeValue = 0, averageTemperature = 0, averageStateOfCharge = 0;                
+                for (int i = 0; i < batteryParameters.Count(); i++)
+                {
+                    sumTemperatureValue += batteryParameters[i].Temperature;
+                    sumStateOfChargeValue += batteryParameters[i].StateOfCharge;
+                }
+                averageTemperature = sumTemperatureValue / batteryParameters.Count();
+                averageStateOfCharge = sumStateOfChargeValue / batteryParameters.Count();
+                batteryStatastics.TemperatureStatastics.TemperatureAverage = averageTemperature;
+                batteryStatastics.SOCStatastics.SOCAverage = averageStateOfCharge;                
+            }
+            else
+            {
+                batteryStatastics.TemperatureStatastics.TemperatureAverage = 0;
```

Comment:
If the parameter list is invalid, you are still returning numbers that look valid?